pkg/auth/audit/audit_test.go

```diff
@@ -126,20 +137,178 @@ func Test_concealSensitiveData(t *testing.T) {
+	req, err := http.NewRequest(http.MethodGet, "/test", nil)
+	a.Require().NoErrorf(err, "Failed to create request: %v", err)
+
+	auditLog, err := newAuditLog(writer, req, sensitiveRegex)
+	a.Require().NoErrorf(err, "Failed to create AuditLog: %v", err)
+
+	const testString = "{\"test\":\"response\"}"
+
+	tests := []struct {
+		name         string
+		respHeader   http.Header
+		respBody     []byte
+		returnCode   int
+		expectedBody string
+		Error        error
+	}{
+		{
+			name:       "invalid Encoding",
+			respHeader: http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"bzip2"}},
+			respBody:   []byte(testString),
+			Error:      fmt.Errorf(""), // any error
+		},
+		{
+			name:         "none encoding",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"none"}},
+			respBody:     []byte(testString),
+			expectedBody: testString,
+		},
+		{
+			name:         "gzip encoding",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"gzip"}},
+			respBody:     a.gzip(testString),
+			expectedBody: testString,
+		},
+		{
+			name:         "deflate encoding",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"deflate"}},
+			respBody:     a.deflate(testString),
+			expectedBody: testString,
+		},
+		{
+			name:         "empty gzip response",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"gzip"}},
+			respBody:     a.gzip("{}"),
+			expectedBody: "{}",
+		},
+		{
+			name:         "empty deflate response",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"deflate"}},
+			respBody:     a.deflate("{}"),
+			expectedBody: "{}",
+		},
+
+		{
+			name:         "invalid response",
+			respHeader:   http.Header{"Content-Type": []string{"application/json"}},
+			respBody:     []byte(""),
+			expectedBody: "",
+		},
+		{
+			name:       "invalid gzip response",
+			respHeader: http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"gzip"}},
+			respBody:   a.gzip(""),
+			Error:      &json.SyntaxError{},
+		},
+		{
+			name:       "invalid deflate response",
+			respHeader: http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"deflate"}},
+			respBody:   a.deflate("Bad Data[]}"),
+			Error:      &json.SyntaxError{},
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
+		a.Run(test.name, func() {
+			// write the test to the audit logger
+			err := auditLog.write(nil, req.Header, test.respHeader, test.returnCode, test.respBody)
+
+			// if we are expecting an error check the error is not nil and the correct type
+			if test.Error != nil {
+				a.Require().Error(err, "Expected error of type '%T'.", test.Error)
+				a.ErrorAsf(test.Error, &err, "Expected error of type '%T'.", test.Error)
```

Comment:
Aren't these basically the same thing? Do we need both? If they serve different purposes, can we use a different error message for one?